resource-managers/yarn/pom.xml

```diff
@@ -126,7 +126,7 @@
     </dependency>
 
      <!--
-    See SPARK-3710. hadoop-yarn-server-tests in Hadoop 2.2 fails to pull some needed
```

Comment:
>  it seemed necessary to keep the hadoop-yarn-server-tests even in 2.6.

Yeah, that one is necessary, it's the other dependencies that were missing from 2.2 (they're proper transitive dependencies in the 2.6 artifact).